core/src/main/java/com/amplifyframework/predictions/models/Feature.java

```diff
@@ -85,7 +85,7 @@ public int compareTo(@Nullable Feature<T> other) {
         if (other == null) {
             return -1;
         }
-        int typeDiff = this.getType().compareToIgnoreCase(other.getType());
+        int typeDiff = this.getTypeAlias().compareToIgnoreCase(other.getTypeAlias());
```

Comment:
When/why would the case differ? Can't we just tighten up the usage is always the same, and then do a literal `equals(...)` check? This is a slippery slope

Reply:
this is comparison implementation. Sure I could have just used `compareTo` instead of `compareToIgnoreCase`, but I figured this would be convenient. Comparing two features of different type is not the correct use-case anyways